src/lib.rs

```diff
@@ -0,0 +1,92 @@
+#[macro_use]
+extern crate dotenv_codegen;
+
+use std::env;
+use futures::executor::block_on;
+use arma_rs::{arma, Extension};
+use mongodb::{Client, Database, options::ClientOptions};
+use mongodb::bson::{doc, Document};
+use mongodb::options::{UpdateOptions};
+use chrono::prelude::{DateTime, Utc};
+use std::time::SystemTime;
+
+use once_cell::sync::OnceCell;
+
+static MONGODB: OnceCell<Database> = OnceCell::new();
+
+#[arma]
+fn init() -> Extension {
+    Extension::build()
+        .command("log", log)
+        .finish()
+}
+
+async fn connect() {
+    if MONGODB.get().is_some() {
+        println!("Connection to DB already present!");
+        return;
+    }
+
+    println!("Connecting to DB!");
+
+    let _url = env::var("EXTENSION_URL").unwrap_or(dotenv!("EXTENSION_URL").to_string());
+    let _db_name = env::var("EXTENSION_DBNAME").unwrap_or(dotenv!("EXTENSION_DBNAME").to_string());
+
+    if let Ok(client_options) = ClientOptions::parse(_url).await {
+        // client_options.app_name = Some("FPArma Server Extension".to_string());
+        if let Ok(client) = Client::with_options(client_options) {
+            let _ = MONGODB.set(client.database(&_db_name[..]));
+            println!("Connected to DB!");
+        }
+    }
+}
+
+pub fn log(id: String, log_level: i32, time: f64, message: String) -> String {
+    // async {
+    //     write_log(&id, log_level, time, &message).await
+    // };
+    let _future = write_log(&id, log_level, time, &message);
+    block_on(_future); // <-- is blocking needed, cannot this be async?
```

Comment:
the previous impl was 
1) receive call to log
2) callback that we're finished
3) write log or w/e to not block arma

Reply:
nvm arma-rs callbacks directly

Reply:
server is waiting until the log function is done. Not ideal, will have to take a look how to call write_log async

Reply:
it is not none blocking